SConstruct

```diff
@@ -507,21 +510,11 @@ if selected_platform in platform_list:
         env.extra_suffix += "." + env["extra_suffix"]
 
     # Environment flags
-    CCFLAGS = env.get("CCFLAGS", "")
-    env["CCFLAGS"] = ""
-    env.Append(CCFLAGS=str(CCFLAGS).split())
```

Comment:
I'm trying to understand why this was being done in the first place. Reading the original value, wiping it, and then appending it as an array, why? The change below is not quite equivalent, as it appends without wiping the value first and doesn't split the string when appending...

might there be a case where one of CCFLAGS/CFLAGS/CXXFLAGS/LINKFLAGS doesn't actually exist or isn't treated as an array, so unless we wipe it and explicitely append as an array it will fail?

Reply:
I wondered too, and tried `env.Append(CCFLAGS=env.get("CCFLAGS", ""))` but I found that it was appending the string twice (`-Wall -Wextra` would turn to `-Wall -Wextra-Wall -Wextra`).

Because the trick is that `en.Append(CCFLAGS=...)` and `env["CCFLAGS"]` are the same variable. So rename helped solve this issue.

Reply:
Is/was scons ever reading these from the OS environment variables?

Reply:
Nope. SCons makes things confusing by calling its own state "Environment" but the only values we read from the OS environment are those:
```
env_base.PrependENVPath("PATH", os.getenv("PATH"))
env_base.PrependENVPath("PKG_CONFIG_PATH", os.getenv("PKG_CONFIG_PATH"))
if "TERM" in os.environ:  # Used for colored output.
    env_base["ENV"]["TERM"] = os.environ["TERM"]
```

```
$ CCFLAGS="-Wbogus" /usr/bin/scons verbose=yes
[Initial build] g++ -o platform/linuxbsd/godot_linuxbsd.linuxbsd.editor.x86_64.o -c -std=gnu++17 -fno-exceptions -pipe -O2 -Wall -Wshadow -Wno-misleading-indentation -Wno-return-type -isystem thirdparty/glad -DTOOLS_ENABLED -DDEBUG_ENABLED -DNDEBUG -DNO_EDITOR_SPLASH -DSOWRAP_ENABLED -DTOUCH_ENABLED -DFONTCONFIG_ENABLED -DALSA_ENABLED -DALSAMIDI_ENABLED -DPULSEAUDIO_ENABLED -D_REENTRANT -DDBUS_ENABLED -DSPEECHD_ENABLED -DXKB_ENABLED -DJOYDEV_ENABLED -DUDEV_ENABLED -DLINUXBSD_ENABLED -DUNIX_ENABLED -D_FILE_OFFSET_BITS=64 -DX11_ENABLED -DVULKAN_ENABLED -DRD_ENABLED -DGLES3_ENABLED -DMINIZIP_ENABLED -DBROTLI_ENABLED -DCLIPPER2_ENABLED -DZSTD_STATIC_LINKING_ONLY -DUSE_VOLK -DVK_USE_PLATFORM_XLIB_KHR -DGLAD_ENABLED -DEGL_ENABLED -DGLAD_GLX_NO_X11 -DHAVE_MNTENT -Ithirdparty/freetype/include -Ithirdparty/libpng -Ithirdparty/volk -Ithirdparty/vulkan -Ithirdparty/vulkan/include -Ithirdparty/zstd -Ithirdparty/zlib -Ithirdparty/clipper2/include -Ithirdparty/brotli/include -Ithirdparty/linuxbsd_headers -Iplatform/linuxbsd -I. platform/linuxbsd/godot_linuxbsd.cpp
```